src/QsCompiler/Transformations/ContentLifting.cs

```diff
@@ -64,17 +63,22 @@ public CallableDetails(QsCallable callable)
 
         public class TransformationState
         {
-            public bool IsValidScope = true;
-            public List<QsCallable> GeneratedOperations = null;
-            public ImmutableArray<LocalVariableDeclaration<NonNullable<string>>> GeneratedOpParams =
+            // ToDo: It should be possible to make these three properties private, 
+            // if we absorb the corresponding logic into LiftBody. 
+            internal bool IsValidScope = true;
+            internal bool ContainsParamRef = false;
+            internal ImmutableArray<LocalVariableDeclaration<NonNullable<string>>> GeneratedOpParams =
                 ImmutableArray<LocalVariableDeclaration<NonNullable<string>>>.Empty;
-            public bool ContainsParamRef = false;
 
-            public CallableDetails CurrentCallable = null;
-            public bool InBody = false;
-            public bool InAdjoint = false;
-            public bool InControlled = false;
-            public bool InWithinBlock = false;
+            internal CallableDetails CurrentCallable = null;
+
+            protected internal bool InBody = false;
+            protected internal bool InAdjoint = false;
+            protected internal bool InControlled = false;
+            protected internal bool InControlledAdjoint = false;
+            protected internal bool InWithinBlock = false;
+
+            public List<QsCallable> GeneratedOperations = null;
```

Comment:
Should this also be internal?

Reply:
Unfortunatly, I think you basically want access to it if you derive from the LiftContent. I don't entirely like that anyone can just add and remove things, but the alternative is to make is protected internal and force people to add a handle to access it when they inherit from it. I am fine either way.

Reply:
going with protected internal